### ISSUE TITLE
In CSS sorting move local properties above nested selectors

### ISSUE THREAD

**Issue by ArekSredzki** [prettydiff/prettydiff#145]
Automatically ordering properties according to a style guide (such as scss-lint's) would be great.

**Comment by prettydiff** [comment]
I have that with the option [objsort](http://prettydiff.com/documentation.xhtml#objsort).  Try it out at http://prettydiff.com/?m=beautify.  Scroll down a bit and look for one of the first options named *Property Sorting, CSS / JavaScript* and check either the second or third radio button.

This option will sort properties and nested selectors.  If this does what you were thinking then simply close the bug.  Otherwise let me know with some examples the kind of option you are looking for.

**Comment by ArekSredzki** [comment]
@prettydiff I didn't realize that this was a feature. I've played around with it for a bit, and it's great, apart from the fact that it put's nested selectors before rules at that level. This interferes with the performance of the new lines commit, but also is against the scss-lint style guide.

**Comment by prettydiff** [comment]
Then moving local properties above child selectors will be the focus of this issue.

**Comment by prettydiff** [comment]
If you wouldn't mind testing this out now: http://prettydiff.com/ignore/testlocation/prettydiff.com.xhtml?m=beautify

I plan to launch v1.12.20 Monday evening at the latest.

**Comment by ArekSredzki** [comment]
It doesn't seem to have fixed the issue.

**Comment by prettydiff** [comment]
Could you provide a sample of code where the issue persists?

**Comment by ArekSredzki** [comment]
Sure thing

Initial:
'''scss
//  Editable Table
.table-control {
    
    &.fa-plus {
        &:hover {
            color: $primary-2;
        }
        
        &:active {
            color: $primary-3;
        }
        color: $primary-1;
    }
    
    &.fa-remove {
        &:hover {
            color: $maroon-0;
        }
        
        &:active {
            color: $maroon-1;
        }
        color: $gray-0;
    }
    cursor: pointer;
    transition: all 0.2s ease-in-out;
}
'''

Output:
'''scss
//  Editable Table
.table-control {
  transition: all 0.2s ease-in-out;
  
  &.fa-plus {
    &:hover {
      color: $primary-2;
    }
    
    &:active {
      color: $primary-3;
    }
    color: $primary-1;
  }
  
  &.fa-remove {
    &:hover {
      color: $maroon-0;
    }
    
    &:active {
      color: $maroon-1;
    }
    color: $gray-0;
  }
  cursor: pointer;
}
'''

**Comment by prettydiff** [comment]
Code updated: http://prettydiff.com/ignore/testlocation/prettydiff.com.xhtml?m=beautify

I found some serious shortcomings in my sort capabilities so I had to change and simplify my approach. It now appears to work on the provided example and I will be throwing other examples at it to see if I am breaking anything.

**Comment by ArekSredzki** [comment]
It only kind of works, if there are some properties in between nested selectors then they stay in place, and if you put the output found above into the beautifier then the transition and cursor properties switch places.
As it is now, all properties from the top are moved down and from the top move up.

**Comment by ArekSredzki** [comment]
Actually, that doesn't seem to be true in all cases, it often puts them in the wrong spot, splitting selectors between the top and bottom of nested ones

**Comment by prettydiff** [comment]
Strange.  I will continue to look at this.

**Comment by prettydiff** [comment]
I think I nailed it now.  I am account for three different types to sort against: *property*, *propvar*, *selector*.  The *propvar* type is something I made up to describe methods, functions, and variables that are not a name/value pair separated by a colon.  I am first sorting by type alphabetically and then if the types are the same between a given two items being compared only them will I sort by name.
